### ISSUE TITLE
Netdata failed to start v1.23.1

### ISSUE THREAD

**Issue by startryst** [netdata/netdata#9475]
CentOS 6, no any change to the metadata, just update to nightly release automatically, below is the tail of the log when starting netdata:
```
2020-07-04 16:29:31: netdata INFO  : MAIN : Metadata log "/var/cache/netdata/dbengine/metadatalog-00000-00001.mlf" loaded (size:155746).
2020-07-04 16:29:31: netdata INFO  : MAIN : Host 'xxx' (at registry as 'xxx') with guid '0905acf2-2591-11ea-8826-aaaa00111ff2' initialized, os 'linux', timezone 'Asia/Shanghai', tags '', program_name 'netdata', program_version 'v1.23.1', update every 1, memory mode dbengine, history entries 3996, streaming disabled (to '' with api key ''), health enabled, cache_dir '/var/cache/netdata', varlib_dir '/var/lib/netdata', health_log '/var/lib/netdata/health/health-log.db', alarms default handler '/usr/libexec/netdata/plugins.d/alarm-notify.sh', alarms default recipient 'root'
2020-07-04 16:29:31: netdata INFO  : MAIN : File '/var/lib/netdata/cloud.d/claimed_id' was found. Setting state to AGENT_CLAIMED.
2020-07-04 16:29:31: netdata ERROR : MAIN : Request to remove RRDVAR 'ipv4.sockstat_frag_sockets.inuse' from index failed. Not Found.
2020-07-04 16:29:31: netdata ERROR : MAIN : RRDVAR: Attempted to delete variable 'ipv4.sockstat_frag_sockets.inuse' from host 'xxx', but it is not found.
2020-07-04 16:29:31: netdata ERROR : MAIN : EOF found in spawn pipe.
Shutting down spawn server event loop.
Shutting down spawn server loop complete.
```

**Comment by startryst** [comment]
tried of reinstallation by the one line script, couldn't fix as well

**Comment by startryst** [comment]
below is another machine(CentOS 7) also encountered the same, attach the tail of the error log:
```
2020-07-04 16:33:57: netdata INFO  : MAIN : Host 'yyy' (at registry as 'yyy') with guid '13c95b7c-2594-11ea-adcb-00163c0ad71f' initialized, os 'linux', timezone 'Asia/Shanghai', tags '', program_name 'netdata', program_version 'v1.23.1', update every 1, memory mode dbengine, history entries 3996, streaming disabled (to '' with api key ''), health enabled, cache_dir '/var/cache/netdata', varlib_dir '/var/lib/netdata', health_log '/var/lib/netdata/health/health-log.db', alarms default handler '/usr/libexec/netdata/plugins.d/alarm-notify.sh', alarms default recipient 'root'
2020-07-04 16:33:57: netdata INFO  : MAIN : File '/var/lib/netdata/cloud.d/claimed_id' was found. Setting state to AGENT_CLAIMED.
2020-07-04 16:33:57: netdata ERROR : MAIN : EOF found in spawn pipe.
Shutting down spawn server event loop.
Shutting down spawn server loop complete.
```

**Comment by stelfrag** [comment]
I am trying to replicate the 

```
2020-07-04 16:29:31: netdata ERROR : MAIN : RRDVAR: Attempted to delete variable 'ipv4.sockstat_frag_sockets.inuse' from host 'xxx', but it is not found.
```

error

**Comment by startryst** [comment]
on another server, which also can't get started, with below errors:
```
2020-07-08 11:04:14: netdata ERROR : MAIN : Request to remove RRDVAR '???' from index failed. Not Found.
2020-07-08 11:04:14: netdata ERROR : MAIN : RRDVAR: Attempted to delete variable '???' from host 'server_hostname', but it is not found.
*** Error in `/usr/sbin/netdata': munmap_chunk(): invalid pointer: 0x000000000eef1d50 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x7f7c4)[0x7f9a25ba47c4]
/usr/sbin/netdata[0x426891]
/usr/sbin/netdata[0x4cf090]
/usr/sbin/netdata[0x4d5251]
/usr/sbin/netdata[0x413bf6]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x7f9a25b47505]
/usr/sbin/netdata[0x4158bd]
======= Memory map: ========
00400000-00585000 r-xp 00000000 fd:00 8463271                            /usr/sbin/netdata
00784000-00785000 r--p 00184000 fd:00 8463271                            /usr/sbin/netdata
00785000-00799000 rw-p 00185000 fd:00 8463271                            /usr/sbin/netdata
```

**Comment by startryst** [comment]
Or is below is the default parameter for the exec?
```
    roles="${1}"               # the roles that should be notified for this event
    args_host="${2}"           # the host generated this event
    unique_id="${3}"           # the unique id of this event
    alarm_id="${4}"            # the unique id of the alarm that generated this event
    event_id="${5}"            # the incremental id of the event, for this alarm id
    when="${6}"                # the timestamp this event occurred
    name="${7}"                # the name of the alarm, as given in netdata health.d entries
    chart="${8}"               # the name of the chart (type.id)
    family="${9}"              # the family of the chart
    status="${10}"             # the current status : REMOVED, UNINITIALIZED, UNDEFINED, CLEAR, WARNING, CRITICAL
    old_status="${11}"         # the previous status: REMOVED, UNINITIALIZED, UNDEFINED, CLEAR, WARNING, CRITICAL
    value="${12}"              # the current value of the alarm
    old_value="${13}"          # the previous value of the alarm
    src="${14}"                # the line number and file the alarm has been configured
    duration="${15}"           # the duration in seconds of the previous alarm state
    non_clear_duration="${16}" # the total duration in seconds this is/was non-clear
    units="${17}"              # the units of the value
    info="${18}"               # a short description of the alarm
    value_string="${19}"       # friendly value (with units)
    # shellcheck disable=SC2034
    # variable is unused, but https://github.com/netdata/netdata/pull/5164#discussion_r255572947
    old_value_string="${20}"   # friendly old value (with units), previously named "old_value_string"
    calc_expression="${21}"    # contains the expression that was evaluated to trigger the alarm
    calc_param_values="${22}"  # the values of the parameters in the expression, at the time of the evaluation
    total_warnings="${23}"     # Total number of alarms in WARNING state
    total_critical="${24}"     # Total number of alarms in CRITICAL state
```

**Comment by mfundul** [comment]
Possibly fixed by https://github.com/netdata/netdata/pull/9503

**Comment by startryst** [comment]
> Possibly fixed by #9503

so how can I get fixed? by re-installing `bash <(curl -Ss https://my-netdata.io/kickstart.sh)`? Then how can I know the version I re-install come swath this fix?

**Comment by mfundul** [comment]
> 
> 
> > Possibly fixed by #9503
> 
> so how can I get fixed? by re-installing `bash <(curl -Ss https://my-netdata.io/kickstart.sh)`? Then how can I know the version I re-installed come with this fix?

Since the Pull Request got merged into master then yes, reinstalling the way you described should suffice.

You can see the version by running `netdata -V`.

**Comment by startryst** [comment]
> > > Possibly fixed by #9503
> > 
> > 
> > so how can I get fixed? by re-installing `bash <(curl -Ss https://my-netdata.io/kickstart.sh)`? Then how can I know the version I re-installed come with this fix?
> 
> Since the Pull Request got merged into master then yes, reinstalling the way you described should suffice.
> 
> You can see the version by running `netdata -V`.

will the stable release include this fix as well?
`bash <(curl -Ss https://my-netdata.io/kickstart.sh) --stable-channel --disable-telemetry`

**Comment by mfundul** [comment]
> 
> 
> > > > Possibly fixed by #9503
> > > 
> > > 
> > > so how can I get fixed? by re-installing `bash <(curl -Ss https://my-netdata.io/kickstart.sh)`? Then how can I know the version I re-installed come with this fix?
> > 
> > 
> > Since the Pull Request got merged into master then yes, reinstalling the way you described should suffice.
> > You can see the version by running `netdata -V`.
> 
> will the stable release include this fix as well?
> `bash <(curl -Ss https://my-netdata.io/kickstart.sh) --stable-channel --disable-telemetry`

Not until the next version.

**Comment by startryst** [comment]
> > > Possibly fixed by #9503
> > 
> > 
> > so how can I get fixed? by re-installing `bash <(curl -Ss https://my-netdata.io/kickstart.sh)`? Then how can I know the version I re-installed come with this fix?
> 
> Since the Pull Request got merged into master then yes, reinstalling the way you described should suffice.
> 
> You can see the version by running `netdata -V`.

I re-installed by `bash <(curl -Ss https://my-netdata.io/kickstart.sh) --no-updates`, and version is `netdata v1.23.1-55-nightly`, but still the netdata couldn't be started, see error log as below:
```
2020-07-09 17:23:18: netdata INFO  : MAIN : RRDSET: chart name 'fping.zptled1_xxx_latency' on host 'yyy' already exists.
2020-07-09 17:23:18: netdata INFO  : MAIN : Collector updated metadata for chart fping.zptled1_xxx_latency
2020-07-09 17:23:18: netdata INFO  : MAIN : Metadata log "/var/cache/netdata/dbengine/metadatalog-00000-00001.mlf" loaded (size:237644).
2020-07-09 17:23:18: netdata INFO  : MAIN : Host 'yyy' (at registry as 'yyy') with guid '69d8cad8-215d-11ea-a9a8-4062310a7cdf' initialized, os 'linux', timezone 'Asia/Shanghai', tags '', program_name 'netdata', program_version 'v1.23.1-55-nightly', update every 1, memory mode dbengine, history entries 3996, streaming disabled (to '' with api key ''), health enabled, cache_dir '/var/cache/netdata', varlib_dir '/var/lib/netdata', health_log '/var/lib/netdata/health/health-log.db', alarms default handler '/usr/libexec/netdata/plugins.d/alarm-notify.sh', alarms default recipient 'root'
2020-07-09 17:23:18: netdata INFO  : MAIN : File '/var/lib/netdata/cloud.d/claimed_id' was found. Setting state to AGENT_CLAIMED.
2020-07-09 17:23:18: netdata ERROR : MAIN : Request to remove RRDVAR ' ©"^O' from index failed. Not Found.
2020-07-09 17:23:18: netdata ERROR : MAIN : RRDVAR: Attempted to delete variable ' ©"^O' from host 'yyy', but it is not found.
*** Error in `/usr/sbin/netdata': munmap_chunk(): invalid pointer: 0x000000000f212590 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x7f7c4)[0x7fe676de67c4]
/usr/sbin/netdata[0x417158]
/usr/sbin/netdata[0x4e33e0]
/usr/sbin/netdata[0x4e87a1]
/usr/sbin/netdata[0x413c46]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x7fe676d89505]
/usr/sbin/netdata[0x41590d]
======= Memory map: ========
00400000-00585000 r-xp 00000000 fd:00 9113883                            /usr/sbin/netdata
00784000-00785000 r--p 00184000 fd:00 9113883                            /usr/sbin/netdata
```

**Comment by ilyam8** [comment]
> 2020-07-09 17:23:18: netdata ERROR : MAIN : Request to remove RRDVAR ' ©"^O' from index failed. Not Found.
2020-07-09 17:23:18: netdata ERROR : MAIN : RRDVAR: Attempted to delete variable ' ©"^O' from host 'yyy', but it is not found.

it looks like we have some problems with variables parsing, i saw a a lot of `Request to remove RRDVAR ... fail` in different issues.

**Comment by stelfrag** [comment]
@startryst would it possible to forward `/var/cache/netdata/dbengine/metadatalog-00000-00001.mlf"` to stelios@netdata.cloud ?

**Comment by startryst** [comment]
> @startryst would it possible to forward `/var/cache/netdata/dbengine/metadatalog-00000-00001.mlf"` to [stelios@netdata.cloud](mailto:stelios@netdata.cloud) ?

sent

**Comment by stelfrag** [comment]
> > @startryst would it possible to forward `/var/cache/netdata/dbengine/metadatalog-00000-00001.mlf"` to [stelios@netdata.cloud](mailto:stelios@netdata.cloud) ?
> 
> sent

Thank you! I received the file and analyzing it..

**Comment by stelfrag** [comment]
I managed to reproduce and identified the offending piece of code. A PR will be issued shortly

**Comment by stelfrag** [comment]
PR #9509 has been merged to master

**Comment by startryst** [comment]
> PR #9509 has been merged to master

when I can re-install the netdata to give a try, i mean if the public available version include the fix.

**Comment by mfundul** [comment]
> 
> 
> > PR #9509 has been merged to master
> 
> when I can re-install the netdata to give a try, i mean if the public available version include the fix.

You can already try via the official method `bash <(curl -Ss https://my-netdata.io/kickstart.sh)`.

**Comment by startryst** [comment]
> > > PR #9509 has been merged to master
> > 
> > 
> > when I can re-install the netdata to give a try, i mean if the public available version include the fix.
> 
> You can already try via the official method `bash <(curl -Ss https://my-netdata.io/kickstart.sh)`.

Fixed, thanks.
